pandas/tests/indexes/test_base.py

```diff
@@ -1034,6 +1034,29 @@ def test_symmetric_difference(self):
         assert tm.equalContents(result, expected)
         assert result.name == 'new_name'
 
+    def test_difference_type(self):
```

Comment:
can you add a test which tries to construct all indices as empty

Reply:
@jreback for a test to construct all indices as empty, is this a test of the public API's (which have to differ based on the specific index class), or a test of the private `Index._shallow_copy([])` ?

For example, `pd.RangeIndex([])` currently fails with `TypeError: RangeIndex(...) must be called with integers, list was passed for start`, which is correct.

Reply:
right you would skip that one. I think there is already a test of construct empty (pretty sure), but make just point it out and make it IS testing everything.